pkg/api/types.go

```diff
@@ -1827,6 +1827,38 @@ func (p *Properties) IsNvidiaDevicePluginCapable() bool {
 	return p.HasNSeriesSKU() && common.IsKubernetesVersionGe(p.OrchestratorProfile.OrchestratorVersion, "1.10.0")
 }
 
+// SetCloudProviderRateLimitDefaults sets default cloudprovider rate limiter config
+func (p *Properties) SetCloudProviderRateLimitDefaults() {
+	if p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket == 0 {
+		if p.HasVMSSAgentPool() {
+			var maxVMSSNodesSupported int
+			if p.HostedMasterProfile != nil {
+				maxVMSSNodesSupported = 100
+			} else {
+				maxVMSSNodesSupported = 250
+			}
+			var rateLimitBucket int
+			for _, profile := range p.AgentPoolProfiles {
+				if profile.AvailabilityProfile == VirtualMachineScaleSets {
+					rateLimitBucket += maxVMSSNodesSupported
+				}
+
+			}
+			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = rateLimitBucket
+		} else {
+			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = DefaultKubernetesCloudProviderRateLimitBucket
+		}
+	}
+	if p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitQPS == 0 {
+		const minQPSToBucketFactor float64 = 0.1
```

Comment:
and @feiskyer @yangl900 the idea here is that we might not want to have the ratio of QPS to bucket size be too extreme; not sure if this is overthinking it...

Reply:
makes sense